code/modules/antagonists/eldritch_cult/eldritch_effects.dm

```diff
@@ -0,0 +1,181 @@
+/obj/effect/eldritch
+	name = "Generic rune"
+	desc = "Weird combination of shapes and symbols etched into the floor itself. The indentation is filled with thick black tar-like fluid."
+	anchored = TRUE
+	icon_state = ""
+	resistance_flags = FIRE_PROOF | UNACIDABLE | ACID_PROOF
+	layer = SIGIL_LAYER
+
+/obj/effect/eldritch/attack_hand(mob/living/user)
+	. = ..()
+	if(.)
+		return
+	try_activate(user)
+
+/obj/effect/eldritch/proc/try_activate(mob/living/user)
+	if(!user.mind.has_antag_datum(/datum/antagonist/e_cult))
+		return
+	flick("[icon_state]_active",src)
+	activate(user)
+
+/obj/effect/eldritch/attacked_by(obj/item/I, mob/living/user)
+	. = ..()
+	if(istype(I,/obj/item/nullrod))
+		qdel(src)
+
+/obj/effect/eldritch/proc/activate(mob/living/user)
+
+	var/datum/antagonist/e_cult/cultie = user.mind.has_antag_datum(/datum/antagonist/e_cult)
+	var/list/knowledge = cultie.get_all_knowledge()
+	var/list/atoms_in_range = list()
+
+	for(var/atom/A in range(1,src))
+		if(istype(A,/mob/living))
+			var/mob/living/L = A
+			if(L.stat != DEAD || L == user) // we only accept corpses, no living beings allowed.
+				continue
+		atoms_in_range += A
+
+	for(var/X in knowledge)
+		var/datum/eldritch_knowledge/EK = X
+		var/list/local_required_atoms = EK.required_atoms
+		if(!local_required_atoms || local_required_atoms.len == 0)
+			continue
+
+		var/list/selected_atoms = list()
+
+		if(!EK.recipe_snowflake_check(atoms_in_range,drop_location(),selected_atoms))
+			continue
+
+		for(var/X1 in local_required_atoms)
+			local_required_atoms[X1] = FALSE
+
+		for(var/X1 in atoms_in_range)
+			var/atom/atom_x1 = X1
+			for(var/X2 in local_required_atoms)
+				var/list/temp_list_X2 = X2
```

Comment:
`A`, `L`, `EK`, `X1`, `X2`, `atom_x1` and `temp_list_X2`, all bad names.
Please use better ones.